File: python/04_004.py
from abc import ABC, abstractmethod

class UserService(ABC):
    @abstractmethod
    def get_user(self, username):
        pass
    
    @abstractmethod
    def create_user(self, user):
        pass
    
    @abstractmethod
    def update_user(self, user):
        pass
    
    @abstractmethod
    def delete_user(self, username):
        pass

File: data structures and algorithms/cpp/04_004.cpp
#include <string>

class User; // Forward declaration

class UserService {
public:
    virtual ~UserService() = default;
    virtual User getUser(const std::string& username) = 0;
    virtual void createUser(const User& user) = 0;
    virtual void updateUser(const User& user) = 0;
    virtual void deleteUser(const std::string& username) = 0;
};